src/controller/retention/faked_controller.go

```diff
@@ -0,0 +1,95 @@
+package retention
```

Comment:
Please use mockery and `make gen_mocks` to generate mocks for the interface.

Add `//go:generate mockery ...` in the `src/testing/controller/controller.go` and run `make gen_mocks` will generate the mocks for the retention controller.